src/events_watcher/mod.rs

```diff
@@ -40,6 +40,7 @@ pub trait EventWatcher {
         store: Arc<Self::Store>,
         contract: &Self::Contract,
         event: Self::Events,
+        log: contract::LogMeta,
```

Comment:
I would highly recommend this to be changed to be a tuple of (event, log), instead of two separate parameters.
```suggestion
        (event, log): (Self::Events, contract::LogMeta)
```
That's better and akin to how ethers-rs API.